flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/SortMergeJoinOperator.java

```diff
@@ -64,7 +64,8 @@
 public class SortMergeJoinOperator extends TableStreamOperator<BaseRow>
 		implements TwoInputStreamOperator<BaseRow, BaseRow, BaseRow>, BoundedMultiInput {
 
-	private final long externalBufferMemory;
+	private final double externalBufferMemRatio;
+	private final long maxExternalBufferMemory;
```

Comment:
why using user configured memory as max external buffer?

Reply:
If we give more than the configured memory, it is not good for performance(except some extreme cases), so we should limit its maximum value.
So why limit it to configured memory? Because if there is a large amount data in one key, users can use this configuration to improve the memory utilization and avoid disk spilling.

Reply:
We already have a relative small default memory for this one, I think this already serve the purpose?

Reply:
> a relative small default memory

Which one?

Reply:
If you think there's code complexity, and you insist, I can remove the biggest limitation. It's just an optimization. We can waste some memory.